src/System.Memory/src/System/Memory.cs

```diff
@@ -246,7 +247,20 @@ public bool TryGetArray(out ArraySegment<T> arraySegment)
         /// allocates, so should generally be avoided, however it is sometimes
         /// necessary to bridge the gap with APIs written in terms of arrays.
         /// </summary>
-        public T[] ToArray() => Span.ToArray();
+        public T[] ToArray()
+        {
+            if (_index < 0)
+            {
+                Span<T> span = ((OwnedMemory<T>)_arrayOrOwnedMemory).AsSpan().Slice(_index & RemoveOwnedFlagBitMask, _length);
+                return span.ToArray();
+            }
+            else
+            {
+                T[] result = new T[_length];
+                Array.Copy((T[])_arrayOrOwnedMemory, result, _length);
+                return result;
```

Comment:
Isn't this missing the offset?

Reply:
(which begs the question whether this is covered by any tests :))

Reply:
Ah yes. good point.

Will fix and add additional tests. The ToArray tests don't have enough scenarios:
https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/Memory/ToArray.cs#L12